Tobemerged in geneweb/master

michelmno opened the pull request:
@a2line @hgouraud when could we have PR #2074 to be merged in geneweb/master ?
but also following commits, and related solved issues:
* 27f2f9eb3e Better encode and escaping HTML functions for books, fix #2054
* 401497658 load forgotten regex.min.js for books
* 6a0a9da52 fix ancestor v1, issue #2083
* 071d6868b update carrousel even if no portrait, fix issue #2078
* 5e295a64d missing etc_prefix for CGI, fix issue #2070

a2line committed:
missing etc_prefix for CGI

fixes #2070

a2line committed:
Better encode and escaping HTML functions for books

* Handles type conversions for encode/escape functions by using intermediate
  variables instead of direct coercion that was causing various errors)
* Apply `move_particle` function only to surnames and domains, not to all books
* Filter out dummy placeholder “?” entries from first name and surname lists
* `Util.uri_encode` replaces `Mutil.encode`, %uri_encode.variable; syntax now
  match the behaviour of escape function making template usage more consistent.
  - Depreciate confusing `html_encode` alias for escape.
  - Depreciate `safe`(_html) unused and not needed on books.
* rewrite substring function to support `%escape.substring;`/`%encode.substring;`
  the breadcrumbs trail now uses only `%foreach;substring.s.n;` (n limited to 10).
* Add a JS function to shorten sup-char values when too many entries left.
* Rewrite `%subtitle;` variable to escape its third value adding a blue highlight.
* Resolves several HTML validation errors: Fix unclosed div elements and mismatched
  tags in list items, properly structure forms elements to maintain valid HTML and
  ensure correct nesting of conditional sections.
* Add Bootstrap tooltips for better accessibility, remove default HTML=true
  option to prevent escaping failure, it can be added where needed.

This should definitively fix #2054.

Above was cherry-picked hgouraud/roglo-7 cid 817fb6ac0, except no changes on
* line  55 about navbar, keep bg-white, not bg-light
* line 190 keep the else leg

hgouraud committed:
update carrousel even if no portrait

Commit:
fix ancestor v1

fixes #2083

lefessan commented on bin/ged2gwb/ged2gwb.ml:
```diff
@@ -2694,8 +2694,23 @@ let find_lev0 (strm__ : _ Stream.t) =
+let open_in_bin_with_bom_check fname =
+  let in_channel_has_bom ic =
+    try
+      Char.code (input_char ic) = 239
```
You can use `input_char ic = '\239'`, which is more readable

michelmno replied:
thank you

hgouraud commented:
Michel, is the "galichet" test base sharable?

hgouraud commented:
Is this PR a replacement for 2074. If yes, close the latter.

michelmno commented:
> Michel, is the "galichet" test base sharable?

Yes it is.

michelmno commented:
> Is this PR a replacement for 2074. If yes, close the latter.

No it is not,  #2074 to be reviewed and merged by itself.

michelmno commented:
I am keeping #2086 to gather cherry picked commits from roglo7 to isolate and test them directly on top of master.

michelmno committed:
load forgoten regex.min.js for books

This is a subset of hgouraud/roglo-7 cid 9fdf07d
candidate for merge directly in master.

a2line commented:
ce travail sert à rien à part nous faire travailler plus je ferme et j'aurais du le faire y'a plusieurs jours

michelmno commented:
> ce travail sert à rien à part nous faire travailler plus je ferme et j'aurais du le faire y'a plusieurs jours

JE ne suis pas d'accord !
puisque la branche sur laquelle vous travaillez hgouraud/roglo-7 ne peux pas être mergé en l'état vu les problèmes que j'ai soulevé dans le PR #2055.
